scripts/build.py

```diff
@@ -609,6 +607,51 @@ def build_files(source, target, file_hashes):
             print e
 
 
+def rebuild_new_files(source_path, target_path, new_files_list, file_hashes):
+    """Minifies list of  new files, removes whitespace from HTML and
+    interpolates paths in HTML to include hashes in source
+    directory and copies it to target.
+
+    Args:
+        source_path: str. Path relative to /oppia directory of directory
+            containing files and directories to be copied.
+        target_path: str. Path relative to /oppia directory of directory where
+            to copy the files and directories.
+        new_files_list: list. List of recently changed file.
+        file_hashes: dict(str, str). Dictionary of file hashes.
+    """
+    for new_file in new_files_list:
+        source_file_path = (source_path + new_file).strip()
+        target_file_path = (target_path + new_file).strip()
+        ensure_directory_exists(target_file_path)
+        print "Minifying %s" % target_file_path
+        minify_func(source_file_path, target_file_path, file_hashes, new_file)
+
+
+def get_new_files_from_directory(source, target):
+    """Compare current's hashed file and newly generated hash
+    and return list of expected new files.
+
+    Args:
+        source: str. Path relative to /oppia where the DEV files are located.
+        target: str. Path relative to /oppia where the BUILD files are located.
+
+    Returns:
+        array(str). List of files expected to be re-hashed.
+    """
+    file_hashes = dict()
+    file_hashes.update(get_file_hashes(source))
+    new_files_list = []
+    for file_name, md5_hash in file_hashes.iteritems():
+        filepath = _insert_hash(target + file_name, md5_hash)
```

Comment:
os.path.join perhaps

Maybe call the return value filepath_with_hash or final_filepath or something like that.

Reply:
Replaced string operation with os.path.join() and renamed `filepath` with `final_filepath`